packages/components/src/custom-select-control/index.js

```diff
@@ -79,6 +79,7 @@ export default function CustomSelectControl( props ) {
 		value: _selectedItem,
 		onMouseOver,
 		onMouseOut,
+		__experimentalShowSelectedHint = false,
```

Comment:
I obviously don't love adding yet another prop 😅 

We've talked before about introducing `renderItem` or something like that which should negate the need for this. I figure let's add the dumb prop for now and come back to this if/when we pick back up on https://github.com/WordPress/gutenberg/pull/41726.

Reply:
This looks like a reasonable compromise that we can remove once we're done reworking `CustomSelectControl`. It should be ok to remove this prop in the near future before it makes its way into WordPress core.

> `CustomSelectControl` is in need of a lot of love!

We are well aware of it! Unfortunately our low capacity and some unforeseen high priority tasks (i.e. fixing the high amount of regressions introduced by the `Popover` refactor) caused some delays. We should be able to resume work on it very soon (maybe we can even start collaborating on it next week, if we find some time)

Reply:
Just a quick request — can we add its associated control in Storybook?

```diff
diff --git a/packages/components/src/custom-select-control/stories/index.js b/packages/components/src/custom-select-control/stories/index.js
index 585c380923..4891bcf210 100644
--- a/packages/components/src/custom-select-control/stories/index.js
+++ b/packages/components/src/custom-select-control/stories/index.js
@@ -8,6 +8,7 @@ export default {
 	component: CustomSelectControl,
 	argTypes: {
 		__next36pxDefaultSize: { control: { type: 'boolean' } },
+		__experimentalShowSelectedHint: { control: { type: 'boolean' } },
 		size: {
 			control: {
 				type: 'radio',
```

Reply:
> (maybe we can even start collaborating on it next week, if we find some time)

Let's!